src/core/components/HeaderBarMenu/HeaderBarMenu.tsx

```diff
@@ -0,0 +1,43 @@
+import './HeaderBarMenu.css';
+import { IconButton } from '../IconButton';
+import { BurgerMenuIcon } from '../../../assets/icons/BurgerMenuIcon';
+import { ProfileIcon } from '../../../assets/icons/ProfileIcon';
+import { useAppDispatch } from '../../../config/redux/hooks';
+import { callHeaderBlockButtonClicked } from '../../redux/coreSlice';
+import { useState } from 'react';
+
+interface HeaderBarMenuProps {
+  companyName: string;
+}
+
+export enum HEADER_BUTTON_ID {
+  burgerMenuButton = 'burger-menu-button',
+  profileMenuButton = 'profile-menu-button',
+}
+
+export const HeaderBarMenu = ({ companyName }: HeaderBarMenuProps) => {
+  const dispatch = useAppDispatch();
+
+  const [buttonID, setButtonID] = useState('');
+
+  const handleButtonClick = (buttonID: HEADER_BUTTON_ID) => () => {
+    dispatch(callHeaderBlockButtonClicked(buttonID));
+    setButtonID(callHeaderBlockButtonClicked(buttonID).payload);
+  };
+
+  return (
+    <div className="root-header-menu">
+      <IconButton
+        handleClick={handleButtonClick(HEADER_BUTTON_ID.burgerMenuButton)}
+        variant={HEADER_BUTTON_ID.burgerMenuButton === buttonID ? 'secondary' : 'primary'}
```

Comment:
Please remove it. We dont need to highlight a button that after click will be always hidden either by a slide-in panel, or by a child page. 
Moreover, looks like with the current implementation, 2 buttons work as a toggle of each other. There is no action to drop the state of both buttons.  It means that as soon as one of the buttons is clicked, there will be always one button active, until page is refreshed or component is rerendered